server/src/main/java/com/octopus/teamcity/opentelemetry/server/BuildOverviewExtensionController.java

```diff
@@ -90,29 +90,50 @@ protected ModelAndView doHandle(@NotNull HttpServletRequest request, @NotNull Ht
                 if (!params.get(PROPERTY_KEY_ENABLED).equals("true"))
                     return null;
 
-                if (!params.get(PROPERTY_KEY_SERVICE).equals(OTELService.HONEYCOMB.getValue()))
+                if (params.get(PROPERTY_KEY_SERVICE).equals(OTELService.CUSTOM.getValue()))
                     return null;
 
                 var traceId = buildStorageManager.getTraceId(build);
                 if (traceId == null)
                     return null;
 
-                var model = mv.getModel();
-                model.put("team", params.get(PROPERTY_KEY_HONEYCOMB_TEAM));
-                model.put("dataset", params.get(PROPERTY_KEY_HONEYCOMB_DATASET));
-                model.put("traceId", traceId);
-
-                //we pad the time to ensure that we get all the spans, just in case we get a slight diff in the
-                //timestamps between the traces and the server start time.
-                //this will also help us see the whole chain better
-                long twoHoursInSeconds = 7200;
-                Date startDate = build.getServerStartDate();
-                model.put("buildStart", (startDate.getTime() / 1000L) - twoHoursInSeconds);
-                var finishDate = build.getFinishDate();
-                model.put("buildEnd", ((finishDate == null ? startDate : finishDate).getTime() / 1000L) + twoHoursInSeconds);
+                if (params.get(PROPERTY_KEY_SERVICE).equals(OTELService.HONEYCOMB.getValue()))
+                    return getHoneycombModelAndView(build, params, traceId);
+                if (params.get(PROPERTY_KEY_SERVICE).equals(OTELService.ZIPKIN.getValue()))
+                    return getZipkinModelAndView(build, params, traceId);
             }
         }
 
+        return null;
+    }
+
+    @NotNull
+    private ModelAndView getZipkinModelAndView(SBuild build, Map<String, String> params, String traceId) {
+        final ModelAndView mv = new ModelAndView(pluginDescriptor.getPluginResourcesPath("buildOverviewZipkinExtension.jsp"));
+
+        var model = mv.getModel();
+        model.put("traceId", traceId);
+        model.put("endpoint", params.get(PROPERTY_KEY_ENDPOINT).replaceAll("/$", ""));
+        return mv;
+    }
+
+    @NotNull
+    private ModelAndView getHoneycombModelAndView(SBuild build, Map<String, String> params, String traceId) {
+        final ModelAndView mv = new ModelAndView(pluginDescriptor.getPluginResourcesPath("buildOverviewHoneycombExtension.jsp"));
+
+        var model = mv.getModel();
+        model.put("team", params.get(PROPERTY_KEY_HONEYCOMB_TEAM));
+        model.put("dataset", params.get(PROPERTY_KEY_HONEYCOMB_DATASET));
+        model.put("traceId", traceId);
+
+        //we pad the time to ensure that we get all the spans, just in case we get a slight diff in the
```

Comment:
Do we need this padding for zipkin too?

Reply:
Nah, as we can link directly to the trace in zipkin